src/components/importwms/ImportWmsDirective.js

```diff
@@ -285,39 +304,79 @@ goog.require('ga_urlutils_service');
           39.37 * 72;
     };
 
+    // Adjust layer extent center according to scale
+    var adjustLayerExtentCenter =
+    function(view, extent, mapSize, maxScaleDenominator) {
+      // We test if the layer extent specified in the
+      // getCapabilities fit the minScale value.
+      var layerExtentScale = getScaleFromExtent(view, extent, mapSize);
+      if (layerExtentScale > maxScaleDenominator) {
+        var layerExtentCenter = ol.extent.getCenter(extent);
+        var factor = layerExtentScale / maxScaleDenominator;
+        var width = ol.extent.getWidth(extent) / factor;
+        var height = ol.extent.getHeight(extent) / factor;
+        extent = [
+          layerExtentCenter[0] - width / 2,
+          layerExtentCenter[1] - height / 2,
+          layerExtentCenter[0] + width / 2,
+          layerExtentCenter[1] + height / 2
+        ];
+        var res = view.constrainResolution(
+          view.getResolutionForExtent(extent,
+          mapSize), 0, -1);
+        view.setCenter(layerExtentCenter);
+        view.setResolution(res);
+        return;
+      }
+    };
+
     // Zoom to layer extent
     var zoomToLayerExtent = function(layer, map) {
       var extent = layer.extent;
       var view = map.getView();
       var mapSize = map.getSize();
 
-      // If a minScale is defined
-      if (layer.MaxScaleDenominator && extent) {
-
-        // We test if the layer extent specified in the
-        // getCapabilities fit the minScale value.
-        var layerExtentScale = getScaleFromExtent(view, extent, mapSize);
-
-        if (layerExtentScale > layer.MaxScaleDenominator) {
-          var layerExtentCenter = ol.extent.getCenter(extent);
-          var factor = layerExtentScale / layer.MaxScaleDenominator;
-          var width = ol.extent.getWidth(extent) / factor;
-          var height = ol.extent.getHeight(extent) / factor;
-          extent = [
-            layerExtentCenter[0] - width / 2,
-            layerExtentCenter[1] - height / 2,
-            layerExtentCenter[0] + width / 2,
-            layerExtentCenter[1] + height / 2
-          ];
+      var mapExtent = map.previousExtent_;
```

Comment:
With my previous comment the `layer.extent` should be set propertly so the center of the extent should be in the swiss extent.  So here you don't need anymore to make all these calculations.